blazar/db/sqlalchemy/api.py

```diff
@@ -2017,7 +2028,8 @@ def resource_properties_list(resource_type):
         query = session.query(
             models.ExtraCapability.capability_name,
             models.ExtraCapability.private,
-            resource_model.capability_value).join(resource_model).distinct()
+            resource_model.capability_value,
+        ).filter(resource_model.deleted.is_(None)).join(resource_model).distinct()
```

Comment:
Agree that we should use that function, it makes refactoring easier/safer. I would imagine join should work out of the box, w/ how I'm guessing it would map to SQL.

Reply:
I initially attempted to do this, but for some reason, this query has its columns prefixed with the model name (so `computehostcapability.deleted` for example), so that function sees it as `!= 'deleted'` and thus returns the original query without filtering the deleted rows.

Reply:
@super-cooper Hm, I think is because of the `join` where it is. What about something like this where it filters first, then joins (no idea if this works):
```
session.query(
            models.ExtraCapability.capability_name,
            models.ExtraCapability.private,
            resource_model.capability_value,
        ).join(model_query(resource_model))
        .distinct()
```

Reply:
I retried using `_read_deleted_filter` and it seems to work. I'm unsure what I was doing wrong before, but I must have been repeatedly typoing something.